Makefile

```diff
@@ -1,323 +1,442 @@
+ifneq ($(origin CATALOG_BASE_IMG), undefined)
+FROM_INDEX_OPT := --from-index $(CATALOG_BASE_IMG)
+endif
+
+# Build a catalog image by adding bundle images to an empty catalog using the operator package manager tool, 'opm'.
+# This recipe invokes 'opm' in 'semver' bundle add mode. For more information on add modes, see:
+# https://github.com/operator-framework/community-operators/blob/7f1438c/docs/packaging-operator.md#updating-your-existing-operator
+.PHONY: catalog-build
+catalog-build: opm ## Build a catalog image.
+	$(OPM) index add --container-tool $(CONTAINER_TOOL) --mode semver --tag $(CATALOG_IMG) --bundles $(BUNDLE_IMGS) $(FROM_INDEX_OPT)
+
+# Push the catalog image.
+.PHONY: catalog-push
+catalog-push: ## Push a catalog image.
+	$(MAKE) docker-push OPERATOR_IMG=$(CATALOG_IMG)
+
+# Recreate all the steps to deploy the operator with OLM using a fully built and pushed catalog image.
+# Requires the following environment variables to be set:
+# - OPERATOR_IMG: The operator image to be used. Default is 'quay.io/openshift/vertical-pod-autoscaler-operator:$OPERATOR_VERSION'.
+# - CATALOG_IMG: The catalog image to be used. Default is 'quay.io/openshift/vertical-pod-autoscaler-operator-catalog:$OPERATOR_VERSION'.
+# - BUNDLE_IMG: The bundle image to be used. Default is 'quay.io/openshift/vertical-pod-autoscaler-operator-bundle:$OPERATOR_VERSION'.
+# Optional:
+# - OPERAND_IMG: The operand image to be used. Default is 'quay.io/openshift/origin-vertical-pod-autoscaler-operator:latest'.
+# - DEPLOY_NAMESPACE: The namespace where the operator will be deployed. Default is 'openshift-vertical-pod-autoscaler'.
+
+## Optionally, the easiest way to pass IMG arguments is to instead set the following environment variables:
+## - IMAGE_TAG_BASE: The base image tag for the operator.
+## - OPERATOR_VERSION: The version of the operator.
+## e.g. make e2e-olm-local IMAGE_TAG_BASE=quay.io/$(USER)/vertical-pod-autoscaler-operator OPERATOR_VERSION=4.17.0 KUBECONFIG=/path/to/kubeconfig
+## This will create OPERATOR_IMG=quay.io/$(IMAGE_TAG_BASE}:4.17.0, BUNDLE_IMG=quay.io/${IMAGE_TAG_BASE}-bundle:4.17.0, and CATALOG_IMG=quay.io/${IMAGE_TAG_BASE}-catalog:4.17.0
+.PHONY: full-olm-deploy
+full-olm-deploy: build docker-build docker-push bundle bundle-build bundle-push catalog-build catalog-push deploy-catalog ## Fully deploy the catalog source that contains the operator. Builds and pushes the operator, bundle, and catalog images. Undeploy with 'make undeploy-catalog'.
+
+# Requires CATALOG_IMG to be set to the catalog image to be used.
+.PHONY: deploy-catalog
+deploy-catalog: create-ns ## Deploy the CatalogSource and OperatorGroup, along with the Operator Subscription.
+	rm -rf $(OLM_OUTPUT_DIR)
+	mkdir -p $(OLM_OUTPUT_DIR)
+	cp -r $(OLM_MANIFESTS_DIR)/* $(OLM_OUTPUT_DIR)
+
+	cd $(OLM_OUTPUT_DIR) && $(KUSTOMIZE) edit set image quay.io/openshift/origin-vertical-pod-autoscaler-operator-catalog=$(CATALOG_IMG)
```

Comment:
I didn't want to edit all these kustomize files when deploying since it would cause a lot of git noise if you are just testing it, so I copy the directory to an output folder like we've done before, and edit everything there.